src/AppInstallerCLICore/Workflows/ManifestComparator.cpp

```diff
@@ -342,7 +342,7 @@ namespace AppInstaller::CLI::Workflow
 
             InapplicabilityFlags IsApplicable(const Manifest::ManifestInstaller& installer) override
             {
-                if (m_requirement == Manifest::ScopeEnum::Unknown || installer.Scope == m_requirement)
+                if (m_requirement == Manifest::ScopeEnum::Unknown ||installer.Scope == m_requirement || DoesInstallerTypeIgnoreScopeFromManifest(installer.InstallerType))

```

Comment:
Also need to change `InstalledScopeFilter` similarly, or upgrade won't work.  Given that you have an upgrade test, it might mean instead that you need to look into why it is working when it shouldn't be...